src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -1707,23 +1710,24 @@ private unsafe void PutAttachments(DocumentsOperationContext context, Document d
                     if (attachment.TryGet(nameof(AttachmentName.Name), out LazyStringValue name) == false ||
                         attachment.TryGet(nameof(AttachmentName.ContentType), out LazyStringValue contentType) == false ||
                         attachment.TryGet(nameof(AttachmentName.Hash), out LazyStringValue hash) == false)
-                        throw new ArgumentException($"The attachment info in missing a mandatory value: {attachment}");
+                        throw new ArgumentException($"The attachment info is missing a mandatory value: {attachment}");
 
-                    var cv = Slices.Empty;
-                    var type = (document.Flags & DocumentFlags.Revision) == DocumentFlags.Revision ? AttachmentType.Revision : AttachmentType.Document;
-
-                    if (isRevision == false && attachmentsStorage.AttachmentExists(context, hash) == false)
+                    if (isRevision == false)
                     {
-                        _documentIdsOfMissingAttachments.Add(document.Id);
+                        if (attachmentsStorage.AttachmentExists(context, hash) == false)
+                            _documentIdsOfMissingAttachments.Add(document.Id);
+
+                        attachmentsStorage.PutAttachmentFromSmuggler(context, document.Id, name, contentType, hash);
+                        continue;
                     }
 
                     using (DocumentIdWorker.GetSliceFromId(_context, document.Id, out Slice lowerDocumentId))
                     using (DocumentIdWorker.GetLowerIdSliceAndStorageKey(_context, name, out Slice lowerName, out Slice nameSlice))
                     using (DocumentIdWorker.GetLowerIdSliceAndStorageKey(_context, contentType, out Slice lowerContentType, out Slice contentTypeSlice))
                     using (Slice.External(_context.Allocator, hash, out Slice base64Hash))
-                    using (type == AttachmentType.Revision ? Slice.From(_context.Allocator, document.ChangeVector, out cv) : (IDisposable)null)
+                    using (Slice.From(_context.Allocator, document.ChangeVector, out Slice cv))
```

Comment:
I prefer keeping it that way because otherwise, we should separate between the cases, and I can't think of a nice way to do so; also, in my opinion, it is more readable like this